### ISSUE TITLE
Uninformative comparison of complex with testthat tolerance

### ISSUE THREAD

**Issue by hadley** [r-lib/waldo#187]
``` r
x <- -0.068237827 - 0i
y <- -0.06823783  + 0i

waldo::compare(x, y, tolerance = testthat::testthat_tolerance())
#> Re(old) != Re(new) but don't know how to show the difference
waldo::compare(x, y)
#> `Re(old)`: -0.068237827
#> `Re(new)`: -0.068237830
```

<sup>Created on 2023-11-29 with [reprex v2.0.2.9000](https://reprex.tidyverse.org)</sup>

Possibly related to #141

**Comment by hadley** [comment]
It's not just complex numbers:

``` r
x <- -0.068237827 
y <- -0.06823783  

waldo::compare(x, y, tolerance = testthat::testthat_tolerance())
#> old != new but don't know how to show the difference
```

<sup>Created on 2023-11-29 with [reprex v2.0.2.9000](https://reprex.tidyverse.org)</sup>

So even more likely to be #141

**Comment by hadley** [comment]
And now that I've fixed #141, this problem has gone away 😄
